File: moderate/column_names.lua
function div(a, b) return math.floor(a/b) end

local o = string.byte('A')
for a in io.lines(arg[1]) do
  a = a - 1
  local d = string.char(a % 26 + o)
  if a >= 26 then
    a = div(a, 26) - 1
    d = string.char(a % 26 + o) .. d
    if a >= 26 then
      a = div(a, 26) - 1
      d = string.char(a % 26 + o) .. d
    end
  end
  print(d)
end

File: moderate/column_names.hs
import System.Environment (getArgs)
import Data.Char (chr, ord)

columnnames  :: Int -> String
columnnames 0 = ""
columnnames x = columnnames (div y 26) ++ [chr (ord 'A' + mod y 26)]
              where y = pred x

main :: IO ()
main = do
    [inpFile] <- getArgs
    input <- readFile inpFile
    putStr . unlines . map (columnnames . read) $ lines input
